### PR TITLE
NEXUS-39240 - open shift operator helm chart

### PR DESCRIPTION
**Pull request by bobotimi**
[NEXUS-39240](https://sonatype.atlassian.net/jira/software/c/projects/NEXUS/boards/47?modal=detail&selectedIssue=NEXUS-39240)

[NEXUS-39240]: https://sonatype.atlassian.net/browse/NEXUS-39240?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: helm-charts/sonatype-nexus-repository/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 56.0.0
```
Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/Chart.yaml:18]: 58?

File: helm-charts/sonatype-nexus-repository/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: 3.56.0
```
Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/Chart.yaml:24]: 3.58.0?

File: helm-charts/sonatype-nexus-repository/tests/local-workdir-daemonset_test.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+      - equal:
+          path: spec.template.spec.volumes
+          value:
+            - name: create-nexus-work-dir-script
+              configMap:
+                name: "nxrm-ha-latest-test-release-create-nexus-workdir-config"
+            - name: host-mnt
+              hostPath:
+                path: /tmp/install
```
Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/tests/local-workdir-daemonset_test.yaml:107]: nit: new line

File: helm-charts/sonatype-nexus-repository/tests/persistentvolume_test.yaml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            - hard
+            - nfsvers=4.1
+
+      - equal:
+          path: spec.nfs.path
+          value: /nexusdata/mountPath
+
+      - equal:
+          path: spec.nfs.server
+          value: 70.71.156.85
```
Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/tests/persistentvolume_test.yaml:167]: nit: new line

File: helm-charts/sonatype-nexus-repository/tests/statefulset_test.yaml [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+      - equal:
+          path: spec.volumeClaimTemplates[0].spec.storageClassName
+          value: "nxrm-ha-latest-test-release-nexusrepo-storage"
+
+      - equal:
+          path: spec.volumeClaimTemplates[0].spec.resources
+          value:
+            requests:
+              storage: 100Gi
```
Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/tests/statefulset_test.yaml:168]: nit:new line

File: helm-charts/sonatype-nexus-repository/templates/statefulset.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  volumeClaimTemplates:
+    - metadata:
+        name: nexusdata
+      spec:
+        accessModes: [ {{ .Values.nexusData.local.enabled | ternary "ReadWriteOnce" ".Values.nexusData.nfs.pvc.accessModes" }} ]
+        storageClassName: {{ .Chart.Name }}-{{ .Chart.Version}}-{{ .Release.Name }}-{{ .Values.nexusData.storageClass.name}}
+        resources:
+          requests:
+            storage: {{ .Values.nexusData.pvc.size }}
+
```
Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/templates/statefulset.yaml:80]: nit: redundant new lines

File: helm-charts/sonatype-nexus-repository/templates/namespaces.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.namespaces.nexusNs }}
+
```
Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/templates/namespaces.yaml:5]: nit: redundant empty line

Comment (oleksiirudyk) [inline helm-charts/sonatype-nexus-repository/templates/namespaces.yaml:5]: nit: redundant empty line

File: helm-charts/sonatype-nexus-repository/templates/statefulset.yaml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+          # See help documentation, these are minimum system requirements
+          resources:
+            {{ toYaml .Values.statefulset.container.resources | nindent 12 }}
+          securityContext:
+            runAsUser: 200
+          imagePullPolicy: {{ .Values.statefulset.container.pullPolicy }}
+          ports:
+            - containerPort: {{ .Values.statefulset.container.containerPort }}
+          env:
+            - name: DATASTORE_CLUSTERED_ENABLED
```
Comment (Mykyta) [inline helm-charts/sonatype-nexus-repository/templates/statefulset.yaml:45]: as discussed add `IS_OPEN_SHIFT_DEPLOYMENT` to env

File: helm-charts/sonatype-nexus-repository/tests/local-workdir-daemonset_test.yaml (hunk 3 above)
Comment (bobotimi) [reply helm-charts/sonatype-nexus-repository/tests/local-workdir-daemonset_test.yaml]: Deleted file

File: helm-charts/sonatype-nexus-repository/templates/local-workdir-daemonset.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+              tail -f /dev/null
+        securityContext:
+          privileged: true 
+      volumes:
+      - name: create-nexus-work-dir-script
+        configMap:
+          name: "{{ .Chart.Name }}-{{ .Chart.Version}}-{{ .Release.Name }}-{{ .Values.workDir.configMapName }}"
+      - name: host-mnt
+        hostPath:
+          path: /tmp/install
```
[the remark this answers is not in the document]
Comment (bobotimi) [reply helm-charts/sonatype-nexus-repository/templates/local-workdir-daemonset.yaml]: Deleted file

File: helm-charts/sonatype-nexus-repository/templates/statefulset.yaml (hunk 8 above)
Comment (bobotimi) [reply helm-charts/sonatype-nexus-repository/templates/statefulset.yaml:39]: Added

File: helm-charts/sonatype-nexus-repository/templates/local-workdir-configmap.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+kind: ConfigMap
+metadata:
+  name: {{ .Chart.Name }}-{{ .Chart.Version}}-{{ .Release.Name }}-{{ .Values.workDir.configMapName }}
+  namespace: {{ .Values.namespaces.nexusNs }}
+data:
+   create-nexus-work-dir.sh: |
+    #!/bin/bash
+    # Make Nexus Repository Manager work directory
+    {{- range $value := until (.Values.statefulset.replicaCount | int )}}
+    mkdir -p /nexus-repo-mgr-work-dir/work{{ $value }}
```
[the remark this answers is not in the document]
Comment (bobotimi) [reply helm-charts/sonatype-nexus-repository/templates/local-workdir-configmap.yaml]: Deleted file

File: helm-charts/sonatype-nexus-repository/Chart.yaml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 58.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: 3.58.1
```
Comment (Mykyta) [inline helm-charts/sonatype-nexus-repository/Chart.yaml:24]: we can change it to 3.59 already

**Review by Mykyta — APPROVED** [review]

**Review by oleksiirudyk — APPROVED** [review]

**Pull request merged**